zmq.hpp

```diff
@@ -87,7 +87,16 @@
 #include <memory>
 #endif
 #ifdef ZMQ_CPP17
+#ifdef __has_include
+#if __has_include(<optional>)
 #include <optional>
+#define ZMQ_HAS_OPTIONAL 1
+#endif
+#if __has_include(<string_view>)
+#define ZMQ_HAS_STRING_VIEW 1
```

Comment:
I was wondering that too,  My understanding of the standard is that there is a constructor for std::string which uses a string_view in the same way as it is being used here,  which means including std::string should be sufficient for this purpose.    I haven't used C++17 mode much and probably won't until next year when we start to switch to C++17,   So I don't have a strong opinion on that, as best as I can tell it is correct either way.

Reply:
Since you test `__has_include(<string_view>)`, it would be consistent to actually include `<string_view>`  in that case, even if it is not necessary for known compilers/library implementations.